test/e2e/conformance/webhook_test.go

```diff
@@ -0,0 +1,230 @@
+			Allowed: true,
+		},
+		ObjectGVK: schema.GroupVersionKind{
+			Group:   "wildwest.dev",
+			Version: "v1alpha1",
+			Kind:    "Cowboy",
+		},
+		T:    t,
+		Lock: sync.Mutex{},
+	}
+
+	testWebhook.StartServer(ctx, cfg, 8090)
+
+	organization := framework.NewOrganizationFixture(t, server)
+	logicalClusters := []logicalcluster.LogicalCluster{
+		framework.NewWorkspaceFixture(t, server, organization, "Universal"),
+		framework.NewWorkspaceFixture(t, server, organization, "Universal"),
+	}
+
+	kubeClusterClient, err := kubernetes.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct client for server")
+	cowbyClients, err := client.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct cowboy client for server")
+
+	//Insall the Cowboy resources into logical clusters
+	for _, logicalCluster := range logicalClusters {
+		t.Logf("Bootstrapping ClusterWorkspace CRDs in logical cluster %s", logicalCluster)
+		apiExtensionsClients, err := apiextensionsclient.NewClusterForConfig(cfg)
+		require.NoError(t, err, "failed to construct apiextensions client for server")
+		crdClient := apiExtensionsClients.Cluster(logicalCluster).ApiextensionsV1().CustomResourceDefinitions()
+		wildwest.Create(t, crdClient, metav1.GroupResource{Group: "wildwest.dev", Resource: "cowboys"})
+	}
+	// Installing webhook into the first workspace.
+	url := "https://localhost:8090/hello"
+	sideEffect := admissionregistrationv1.SideEffectClassNone
+	webhook := &admissionregistrationv1.ValidatingWebhookConfiguration{
+		TypeMeta:   metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{Name: "test-webhook"},
+		Webhooks: []admissionregistrationv1.ValidatingWebhook{{
+			Name: "test-webhook.cowboy.io",
+			ClientConfig: admissionregistrationv1.WebhookClientConfig{
+				URL:      &url,
+				CABundle: cfg.CAData,
+			},
+			Rules: []admissionregistrationv1.RuleWithOperations{{
+				Operations: []admissionregistrationv1.OperationType{
+					admissionregistrationv1.Create,
+				},
+				Rule: admissionregistrationv1.Rule{
+					APIGroups:   []string{"wildwest.dev"},
+					APIVersions: []string{"v1alpha1"},
+					Resources:   []string{"cowboys"},
+				},
+			}},
+			SideEffects:             &sideEffect,
+			AdmissionReviewVersions: []string{"v1"},
+		}},
+	}
+	_, err = kubeClusterClient.Cluster(logicalClusters[0]).AdmissionregistrationV1().ValidatingWebhookConfigurations().Create(ctx, webhook, metav1.CreateOptions{})
+	require.NoError(t, err, "failed to add validating webhook configurations")
+	cowboy := v1alpha1.Cowboy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testing",
+			Namespace: "default",
+		},
+		Spec: v1alpha1.CowboySpec{},
+	}
+	require.Eventually(t, func() bool {
+		_, err = cowbyClients.Cluster(logicalClusters[1]).WildwestV1alpha1().Cowboys("default").Create(ctx, &cowboy, metav1.CreateOptions{})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return false
+		}
+		return true
+
+	}, 10*time.Second, 2*time.Second)
```

Comment:
We typically use `wait.ForeverTestTimeout` for the timeout, and a shorter interval (e.g. 100ms)

Reply:
Also, whitespace